### PR TITLE
[10.0] Add module account_invoice_payment_report

### PR DESCRIPTION
**Pull request by grindtildeath**
Backport v12.0 account invoice report with payments

### CODE REVIEW

File: account_invoice_payment_report/models/account_invoice.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            if payment_currency_id and payment_currency_id == self.currency_id:
+                amount_to_show = amount_currency
+            else:
+                currency = payment.company_id.currency_id
+                ctx_date = self.date or fields.Date.today()
+                amount_to_show = currency.with_context(date=ctx_date).compute(
+                    amount, self.currency_id
+                )
+            if float_is_zero(
+                amount_to_show, precision_rounding=self.currency_id.rounding
```
Comment (i-vyshnevska) [inline account_invoice_payment_report/models/account_invoice.py:86]:
```suggestion
                amount_to_show, precision_rounding=currency_id.rounding
```

File: account_invoice_payment_report/models/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                                p.currency_id
+                                == payment.matched_credit_ids[0].currency_id
+                                for p in payment.matched_credit_ids
+                            ]
+                        )
+                        and payment.matched_credit_ids[0].currency_id
+                        or False
+                    )
+            # get the payment value in invoice currency
+            if payment_currency_id and payment_currency_id == self.currency_id:
```
Comment (i-vyshnevska) [inline account_invoice_payment_report/models/account_invoice.py:77]:
```suggestion
            if payment_currency_id and payment_currency_id == currency_id:
```

File: account_invoice_payment_report/models/account_invoice.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                amount_to_show = amount_currency
+            else:
+                currency = payment.company_id.currency_id
+                ctx_date = self.date or fields.Date.today()
+                amount_to_show = currency.with_context(date=ctx_date).compute(
+                    amount, self.currency_id
```
Comment (i-vyshnevska) [inline account_invoice_payment_report/models/account_invoice.py:83]:
```suggestion
                    amount, currency_id
```

File: account_invoice_payment_report/models/account_invoice.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                )
+                amount_currency = sum(
+                    [
+                        p.amount_currency
+                        for p in payment.matched_debit_ids
+                        if p.debit_move_id in self.move_id.line_ids
+                    ]
+                )
+                if payment.matched_debit_ids:
+                    payment_currency_id = (
```
Comment (jcoux) [inline account_invoice_payment_report/models/account_invoice.py:38]: The value returned by your code is a currency (record set), not the id.
s/payment_currency_id/payment_currency

File: account_invoice_payment_report/models/account_invoice.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                )
+                amount_currency = sum(
+                    [
+                        p.amount_currency
+                        for p in payment.matched_credit_ids
+                        if p.credit_move_id in self.move_id.line_ids
+                    ]
+                )
+                if payment.matched_credit_ids:
+                    payment_currency_id = (
```
Comment (jcoux) [inline account_invoice_payment_report/models/account_invoice.py:65]: The value returned by your code is a currency (record set), not the id.
s/payment_currency_id/payment_currency

File: account_invoice_payment_report/models/account_invoice.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    _inherit = 'account.invoice'
+
+    # Backport of odoo v12.0 method
+    def _get_payments_vals(self):
+
+        self.ensure_one()
+        if not self.payment_move_line_ids:
+            return []
+        payment_vals = []
+        currency_id = self.currency_id
```
Comment (jcoux) [inline account_invoice_payment_report/models/account_invoice.py:19]: `self.currency_id` returned a currency (record set), not the id.
s/currency_id/currency

File: account_invoice_payment_report/models/account_invoice.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+                            ]
+                        )
+                        and payment.matched_credit_ids[0].currency_id
+                        or False
+                    )
+            # get the payment value in invoice currency
+            if payment_currency_id and payment_currency_id == self.currency_id:
+                amount_to_show = amount_currency
+            else:
+                currency = payment.company_id.currency_id
```
Comment (jcoux) [inline account_invoice_payment_report/models/account_invoice.py:80]: If you rename the variable `currency_id` as proposed, you need to have a different variable name here.
s/currency/company_currency

File: account_invoice_payment_report/views/report_invoice.xml [hunk 8]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+
+    <template id="report_invoice_document_with_payments" inherit_id="account.report_invoice_document" primary="True">
+        <xpath expr="//span[@t-field='o.amount_untaxed']/ancestor::table" position="inside">
+            <t t-set="payments_vals" t-value="o._get_payments_vals()"/>
+            <t t-foreach="payments_vals" t-as="payment_vals">
+                <tr>
+                    <td>
+                        <i class="oe_form_field text-right oe_payment_label">Paid on <t t-esc="payment_vals['date']"/></i>
```
Comment (jcoux) [inline account_invoice_payment_report/views/report_invoice.xml:10]:
```suggestion
                        <i class="oe_form_field text-right oe_payment_label">Paid on <t t-esc="payment_vals['date']"/>
```

**Review by jcoux — CHANGES_REQUESTED** [review]

File: account_invoice_payment_report/views/report_invoice.xml (hunk 8 above)
Comment (grindtildeath) [reply account_invoice_payment_report/views/report_invoice.xml:10]: `<i>`element must be closed...

**Comment by grindtildeath** [comment]
@i-vyshnevska @jcoux This is a backport from v12.0, I'd like to avoid messing with odoo code only for var names, since it's working now with their code...

**Comment by jcoux** [comment]
@grindtildeath Maybe in this case we want to change also the version 12.0 of the code?

**Comment by grindtildeath** [comment]
@jcoux You mean open a PR on odoo for this ? I guess there would be lots of PRs to open for such changes :smile:

File: account_invoice_payment_report/views/report_invoice.xml (hunk 8 above)
Comment (jcoux) [reply account_invoice_payment_report/views/report_invoice.xml:10]: You're right!

**Review by jcoux — APPROVED** [review]
I didn't see that the original 12.0 code was an odoo code, so it's ok for me like this :+1:

**Review by i-vyshnevska — APPROVED** [review]

**Review by TDu — APPROVED** [review]

File: account_invoice_payment_report/models/account_invoice.py (hunk 6 above)
Comment (tschanzt) [reply account_invoice_payment_report/models/account_invoice.py:19]: i would just drop that line

**Review by tschanzt — APPROVED** [review]
lgtm

**Comment by grindtildeath** [comment]
This should probably be moved to https://github.com/OCA/account-invoice-reporting

**Comment by AaronHForgeFlow** [comment]
Agree to move this to account-invoice-reporting repo

**Comment by grindtildeath** [comment]
Moved to https://github.com/OCA/account-invoice-reporting/pull/122

**Pull request closed without merging**